articles/getting-started-with-matlab/README.md

```diff
@@ -0,0 +1,207 @@
+
+Now go and execute `values(1:3, 2:4)` yourself. What will be the result?
+#### Zeros, Ones and Eye
+When you use `zeros` which is a built-in function, it makes every value equal to zero. For example, if you write `a = zeros(4:5)` then the result will be four rows and five columns of zeros:
+```matlab
+a =
+
+     0     0     0     0     0
+     0     0     0     0     0
+     0     0     0     0     0
+     0     0     0     0     0
+```
+`Ones` use the same method above, it makes every value equal to one. Try `b = ones(5:6)` and see the result yourself.
+
+`Eye` is a built-in function as well, it makes every value on the main diagonal equal to one and zeros elsewhere. Example:
+```matlab
+c = eye(4)
+```
+The result is:
+```matlab
+c =
+
+     1     0     0     0
+     0     1     0     0
+     0     0     1     0
+     0     0     0     1
+```
+#### Linspace 
+Linspace function generates linearly spaced vectors. The difference between colon operator and linspace is that the second one gives us control over the points. For example, if you write `linspace(1,10,10)` then the result will be:
+```matlab
+ans =
+
+     1     2     3     4     5     6     7     8     9    10
+```
+But if you change the last value to 8 instead of 10 then the result will be:
+```matlab
+ans =
+
+    1.0000    2.2857    3.5714    4.8571    6.1429    7.4286    8.7143   10.0000
+```
+
+#### M-files
+Entering commands at MATLAB prompt is not practical when the problems will be complicated. The solution will be using M-files which is user-defined commands that normally have input & output and the script will be executed in sequence when it is called.
+
+It is saved with the extension ".m".
+
+Now let's create a function. First of all, you need to press on *New Script*:
+
+![](/engineering-education/getting-started-with-matlab/1.PNG)
+
+Second, press on *New* then *Function*:
+
+![](/engineering-education/getting-started-with-matlab/2.PNG)
+
+It is that easy! Now let's create a circle area function by:
+```matlab
+function [area] = areafunction(r)
+%UNTITLED3 Summary of this function goes here
+%   Detailed explanation goes here
+area = pi*r*r;
+end
+```
+
+After that, you have to save the function with the extension ".m" as we mentioned above.
+Finally, write `areafunction(2)` in the command window and the result will be:
+```matlab
+ans =
+
+   12.5664
+```
+
+#### Plotting
+If you are familiar with the Matplotlib plotting library for Python programming language, it will be very easy for you to follow this section. However, we will explain it simply as possible.
+
+To plot any graph we need to define the range of *X* and *Y* values. Then to call the plot command which is `plot(X,Y)`.
+
+Now let's move on to a complex example. We are going to create the following figure:
+
+![sin and cos waves](/engineering-education/getting-started-with-matlab/sin-cos.jpg)
+
+First of all, let's define the t (a variable indicating time).
```

Comment:
Don't use first of all a lot. It makes the reader lose interest. First of all should be used carefully, once or twice per article.

Reply:
define t... the is not needed.